files/enc.rb

```diff
@@ -35,9 +35,33 @@ def puppetuser
   SETTINGS[:puppetuser] || 'puppet'
 end
 
-def stat_file(certname)
-  FileUtils.mkdir_p "#{puppetdir}/yaml/foreman/"
-  "#{puppetdir}/yaml/foreman/#{certname}.yaml"
+def fact_extension
+  SETTINGS[:fact_extension] || 'yaml'
+end
+
+def fact_directory
+  if fact_extension == 'yaml'
+    File.join(puppetdir, fact_extension, 'facts')
+  else
+    File.join(puppetdir, 'server_data', 'facts')
+  end
+end
+
+def fact_file(certname)
+  File.join(fact_directory, "#{certname}.#{fact_extension}")
+end
+
+def fact_files
+  Dir[File.join(fact_directory, "*.#{fact_extension}")]
+end
+
+def certname_from_filename(filename)
+  File.basename(filename, ".#{fact_extension}")
+end
+
+def stat_file(certname, extension='yaml')
+  FileUtils.mkdir_p "#{puppetdir}/#{extension}/foreman/"
+  "#{puppetdir}/#{extension}/foreman/#{certname}.#{extension}"
 end
```

Comment:
why isn't this using `fact_extension`?

Reply:
ah, because it's not for files from Pupept…

Reply:
Yes, that also tricked me a bit.

Reply:
Then I am not sure why we add the `extension` param here at all?

Reply:
Mostly because I started to think in the same way you did but you're right that not adding it would make it clearer. Especially if the rest of the code can't handle a different extension.